viewing_party/party.py

```diff
@@ -1,23 +1,128 @@
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {"title": title, "genre": genre, "rating": rating}
+    else:
+        return None
+
+    
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watched_movie = user_data['watched']
+    movie_to_watchlist = user_data['watchlist']
+
+    for item in movie_to_watchlist:
+        if title == item['title']:
+            watched_movie.append(item)
+            movie_to_watchlist.remove(item)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_ratings = 0
+    number_movies = len(user_data["watched"])
+
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    
+    for i in range(len(user_data["watched"])):
+            total_ratings += user_data["watched"][i]["rating"]
+            average_rating = total_ratings / number_movies
+    return average_rating

```

Comment:
This looks good overall. One thing to think about however is whether a "for i in range" loop is better than a simple "for (?) in (?)" loop! While user_data["watched"] is a list of movie dictionaries, a "for (?) in (?) loop" will still go through each item there! This would allow us to avoid using the extra [i] as part of our statement on line 41.